Source/WebCore/css/typedom/transform/CSSRotate.cpp

```diff
@@ -70,6 +74,53 @@ ExceptionOr<Ref<CSSRotate>> CSSRotate::create(Ref<CSSNumericValue> angle)
         WTFMove(angle)));
 }
 
+ExceptionOr<Ref<CSSRotate>> CSSRotate::create(CSSFunctionValue& cssFunctionValue)
+{
+    auto makeRotate = [&](const Function<ExceptionOr<Ref<CSSRotate>>(Vector<RefPtr<CSSNumericValue>>)>& create) -> ExceptionOr<Ref<CSSRotate>> {
+        Vector<RefPtr<CSSNumericValue>> components;
+        for (auto componentCSSValue : cssFunctionValue) {
+            auto valueOrException = CSSStyleValueFactory::reifyValue(componentCSSValue);
+            if (valueOrException.hasException())
+                return valueOrException.releaseException();
+            if (!is<CSSNumericValue>(valueOrException.returnValue()))
+                return Exception { TypeError, "Expected a CSSNumericValue."_s };
+            components.append(downcast<CSSNumericValue>(valueOrException.releaseReturnValue().ptr()));
+        }
+        return create(components);
+    };
+
+    switch (cssFunctionValue.name()) {
+    case CSSValueRotateX:
+        return makeRotate([](Vector<RefPtr<CSSNumericValue>> components) {
+            ASSERT(components.size() == 1);
+            return CSSRotate::create(CSSNumericFactory::number(1), CSSNumericFactory::number(0), CSSNumericFactory::number(0), *components[0]);
+        });
+    case CSSValueRotateY:
+        return makeRotate([](Vector<RefPtr<CSSNumericValue>> components) {
+            ASSERT(components.size() == 1);
+            return CSSRotate::create(CSSNumericFactory::number(0), CSSNumericFactory::number(1), CSSNumericFactory::number(0), *components[0]);
+        });
+    case CSSValueRotateZ:
+        return makeRotate([](Vector<RefPtr<CSSNumericValue>> components) {
+            ASSERT(components.size() == 1);
+            return CSSRotate::create(CSSNumericFactory::number(0), CSSNumericFactory::number(0), CSSNumericFactory::number(1), *components[0]);
+        });
+    case CSSValueRotate:
+        return makeRotate([](Vector<RefPtr<CSSNumericValue>> components) {
+            ASSERT(components.size() == 1);
+            return CSSRotate::create(*components[0]);
+        });
+    case CSSValueRotate3d:
+        return makeRotate([](Vector<RefPtr<CSSNumericValue>> components) {
+            ASSERT(components.size() == 4);
```

Comment:
Ditto here.  check components is the right size and the nullness is not unexpected, then assert_not_reached and return an exception if anything goes horribly wrong.

Reply:
Addressed in 6b8ab023c1ea.